mesop/components/html/html.ts

```diff
@@ -16,12 +17,52 @@ export class HtmlComponent {
   @Input({required: true}) type!: Type;
   @Input() key!: Key;
   @Input() style!: Style;
+  @ViewChild('iframe', {read: ElementRef}) iframe!: ElementRef;
   private _config!: HtmlType;
+  isSandboxed!: boolean;
+  html!: string;
 
   ngOnChanges() {
     this._config = HtmlType.deserializeBinary(
       this.type.getValue() as unknown as Uint8Array,
     );
+    this.isSandboxed = this._config.getMode() === 'sandboxed';
+    if (!this.isSandboxed) {
+      return;
+    }
+    const previousHtml = this.html;
+    this.html = this._config.getHtml()!;
+    // Reload iframe if the HTML has changed.
+    if (
+      this.html !== previousHtml &&
+      this.iframe &&
+      this.iframe.nativeElement
+    ) {
+      this.loadIframe();
+    }
+  }
+
+  ngAfterViewInit() {
+    this.loadIframe();
+  }
+
+  loadIframe(): void {
+    if (!this.iframe) {
+      if (this.isSandboxed) {
+        console.warn(
+          'iframe element in Mesop html component unexpectedly not found',
+        );
+      }
+      return;
+    }
+    const iframe = this.iframe.nativeElement as HTMLIFrameElement;
+    setIframeSrc(iframe, '/sandbox_iframe.html', {allowSameOrigin: false});
+    iframe.onload = () => {
```

Comment:
@richard-to wondering whether if I should automatically resize the iframe if the style height is not set (vs. the browser default which is 400px for iframe height, I think). I think most people would want/expect the html component to display all the contents and not have a scrollbar. WDYT?

Reply:
Yeah, if you're able to render the iframe to the correct size of the content without a scrollbar, I would agree that makes a lot of sense since one would expect it to work as if they were rendering HTML. So would be slightly annoying to see it rendered to 400px.

Reply:
done - I've automatically resized the iframe